doc/SLSKPROTOCOL.md

```diff
@@ -2777,57 +2777,59 @@ A peer sends this message when it has a file search match. The token is taken fr
+                - <ins>bitrate</ins> *Kbps*
+                - <ins>duration</ins> *Seconds*
+                - <ins>vbr</ins> 0 *or* 1 *(is Variable Bit Rate)*
     5.  **bool** <ins>slotfree</ins>
     6.  **uint** <ins>avgspeed</ins>
     7.  **uint64** <ins>queue length</ins>
-    8.  **uint** <ins>private results size</ins> *number of privately shared results*
-    9.  Iterate for number of privately shared results
+    8.  **uint** <ins>number of privately shared results</ins>
+    9.  Iterate for <ins>number of privately shared results</ins>
         1.  **uchar** 1
         2.  **string** <ins>filename</ins>
         3.  **uint64** <ins>size</ins>
         4.  **string** <ins>ext</ins>
-        5.  **uint** <ins>attribute size</ins>
-        6.  Iterate <ins>number of attributes</ins>
-            1.  **uint** <ins>place in
-                attributes</ins>
+        5.  **uint** <ins>number of attributes</ins> 0 *or* 3
+        6.  Iterate for <ins>number of attributes</ins>
+            1.  **uint** <ins>place in attributes</ins> 0 *(bitrate)*;  1 *(duration)*;  2 *(vbr)*
             2.  **uint** <ins>attribute</ins>
+                - <ins>bitrate</ins> *Kbps*
+                - <ins>duration</ins> *Seconds*
+                - <ins>vbr</ins> 0 *or* 1 *(is Variable Bit Rate)*
   - Receive
     1.  decompress
     2.  **string** <ins>username</ins>
     3.  **uint** <ins>token</ins>
-    4.  **uint** <ins>results size</ins> *number of results*
+    4.  **uint** <ins>number of results</ins>
     5.  Iterate for <ins>number of results</ins>
         1.  **string** <ins>filename</ins>
         2.  **uint64** <ins>size</ins>
         3.  **string** <ins>ext</ins>
-        4.  **uint** <ins>number of attributes</ins>
-        5.  Iterate <ins>number of attributes</ins>
-            1.  **uint** <ins>place in
-                attributes</ins>
+        4.  **uint** <ins>number of attributes</ins> 2 *or* 3 *(depending on <ins>ext</ins>)*;  0 *(no metadata)*
+        5.  Iterate for <ins>number of attributes</ins>
+            1.  **uint** <ins>place in attributes</ins> *(Order varies between clients and <ins>ext</ins>)*
             2.  **uint** <ins>attribute</ins>
     6.  **bool** <ins>slotfree</ins>
     7.  **uint** <ins>avgspeed</ins>
     8.  **uint64** <ins>queue length</ins>
-    9.  **uint** <ins>private results size</ins> *number of privately shared results*
+    9.  **uint** <ins>number of privately shared results</ins>
     10.  Iterate for <ins>number of privately shared results</ins>
          1.  **string** <ins>filename</ins>
          2.  **uint64** <ins>size</ins>
          3.  **string** <ins>ext</ins>
-         4.  **uint** <ins>number of attributes</ins>
-         5.  Iterate <ins>number of attributes</ins>
-             1.  **uint** <ins>place in
-                 attributes</ins>
+         4.  **uint** <ins>number of attributes</ins> 2 *or* 3 *(depending on <ins>ext</ins>)*;  0 *(no metadata)*
```

Comment:
This can actually be 1 in rare cases (sent from SoulseekQt), and I believe the attribute could either be a bitrate or sample rate, but I haven't analyzed how SoulseekQt differentiates these two yet.

Reply:
How curious, I suppose it is unlikely that a sample rate would be less than 4096, and bitrates are usually less than that.